### PR TITLE
10.3.0 consolidated block stringnull

### PR DESCRIPTION
**Pull request by Expressionless**
This marks answers with value "null" (string) as unanswered, and stops iteration over the fields if it encounters one unanswered field. Previously a bug occurs that if it comes across an unanswered field it would keep iterating anyway, which gave the opportunity if it to decide everything was answered if it later came across another answered field

### CODE REVIEW

File: qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java [hunk 1]
```diff
@@ -474,7 +474,7 @@ public Map<String, Ask> updateDependentAsks(BaseEntity target, Definition defini
 	 * @param baseEntity The BaseEntity to check against
 	 * @return Boolean
 	 */
-	public static Boolean mandatoryFieldsAreAnswered(Map<String, Ask> map, BaseEntity baseEntity) {
+	public static boolean mandatoryFieldsAreAnswered(Map<String, Ask> map, BaseEntity baseEntity) {
 
 		// find all the mandatory booleans
 		Boolean answered = true;
```
Comment (varunshastry) [inline qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java:480]: This could be changed to primitive too.

**Review by varunshastry — CHANGES_REQUESTED** [review]

File: qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java [hunk 2]
```diff
@@ -496,18 +496,41 @@ public static Boolean mandatoryFieldsAreAnswered(Map<String, Ask> map, BaseEntit
...
+				answered = acceptableAnswer(value);
 
-			String resultLine = (mandatory ? "[M]" : "[O]") + " : " + attributeCode + " : " + value;
-			log.debug("===> " + resultLine);
+			if(!answered) {
+				log.debug("Found unanswered mandatory: " + attributeCode);
+				break;
+			}
+
+			String resultLine = (readonly ? "[R]" : "") + (mandatory ? "[M]" : "[O]") + " : " + attributeCode + " : " + value;
```
Comment (jasperRob) [inline qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java]: If it is readonly, then we are not expecting any answer, so we should be ignoring them.

**Review by jasperRob — CHANGES_REQUESTED** [review]

File: qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java (hunk 2 above)
Comment (Expressionless) [reply qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java]: Apologies, was starting to scratch my head a bit about why this one answer was being let through supposedly

File: qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java (hunk 1 above)
Comment (Expressionless) [reply qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java:480]: This is true

**Review by jasperRob — APPROVED** [review]

**Review by varunshastry — APPROVED** [review]

**Pull request merged**